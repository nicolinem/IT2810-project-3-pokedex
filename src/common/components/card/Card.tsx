import React from "react";
import { useNavigate } from "react-router-dom";
import { getImageUrl } from "../../../api/utils/match.utils";
import { matchType, Pokemon } from "../../../types/pokemon.utils";
import { Type } from "../Type";

type Props = {
  children?: React.ReactNode;
  className?: string;
  pokemon: Pokemon;
};

const Card: React.FC<Props> = ({ children, pokemon }) => {
  const navigate = useNavigate();

  const handleClick = () => {
    navigate("/pokemon/" + pokemon.pokemonID);
  };

  return (
    <div
      onClick={handleClick}
      className="w-full flex flex-col content-start justify-start items-center px-10 py-4 bg-white border-gray-400 rounded-lg shadow-lg lg:max-w-sm hover:cursor-pointer hover:bg-[#b8c1d4]"
    >
      <img
        className="object-cover w-full max-w-fit"
        src={getImageUrl(pokemon.pokemonID)}
        alt="image of pokemon"
        height={15}
        width={15}
      /> 
      <div className="justify-self-center flex-col p-4 max-w-fit w-30 h-30">
        <p className="mb-2 text-sm font-semibold leading-normal tracking-widest text-center uppercase xxs:text-xxs xs:text-xs sm:text-sm xl:text-lg lg:text-sm text-transform:">
          {pokemon.name}
        </p>
        <div className="space-y-2 grid top-0 place-items-center tracking-widest ">
          {pokemon.type2 === "" ? (
            <Type type={matchType(pokemon.type1)}></Type>
          ) : (
            <>
              <div className="space-y-2 justify-center">
                <Type type={matchType(pokemon.type1)} />
                <Type type={matchType(pokemon.type2)} />
              </div>
            </>
          )}
        </div>
      </div>
    </div>
  );
};

export default Card;
