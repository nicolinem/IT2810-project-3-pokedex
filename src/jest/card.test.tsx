import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Card from "../common/components/card/Card";

describe("Card", () => {
  it("Snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Card
            pokemon={{
              attack: 65,
              defence: 65,
              height: 7,
              hp: 45,
              imageUrl:
                "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
              name: "bulbasaur",
              pokemonID: 1,
              sp_attack: 49,
              sp_defence: 65,
              speed: 45,
              type1: "grass",
              type2: "poison",
              weight: 69,
            }}
          ></Card>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
