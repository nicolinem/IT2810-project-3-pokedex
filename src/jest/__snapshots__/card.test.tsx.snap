// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot 1`] = `
<div
  className="w-full flex flex-col content-start justify-start items-center px-10 py-4 bg-white border-gray-400 rounded-lg shadow-lg lg:max-w-sm hover:cursor-pointer hover:bg-[#b8c1d4]"
  id="card"
  onClick={[Function]}
>
  <img
    alt="image of pokemon"
    className="object-cover w-full max-w-fit"
    height={15}
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
    width={15}
  />
  <div
    className="flex-col p-4 justify-self-center max-w-fit w-30 h-30"
  >
    <p
      className="mb-2 text-sm font-semibold leading-normal tracking-widest text-center uppercase xxs:text-xxs xs:text-xs sm:text-sm xl:text-lg lg:text-sm text-transform:"
    >
      bulbasaur
    </p>
    <div
      className="top-0 grid space-y-2 tracking-widest place-items-center "
    >
      <div
        className="justify-center space-y-2"
      >
        <div
          className="max-w-fit "
        >
          <div
            className="flex justify-around text-black text-sm font-bold md:max-lg:flex rounded-full w-24 h-7 pt-1 xxs:h-4 xs:h-5 sm:h-6 lg:h-7 xxs:w-14 xs:w-16 sm:w-20 lg:w-24 xxs:text-xxs xs:text-xs sm:text-sm lg:text-lg bg-[#78C850]"
          >
            grass
          </div>
        </div>
        <div
          className="max-w-fit "
        >
          <div
            className="flex justify-around text-black text-sm font-bold md:max-lg:flex rounded-full w-24 h-7 pt-1 xxs:h-4 xs:h-5 sm:h-6 lg:h-7 xxs:w-14 xs:w-16 sm:w-20 lg:w-24 xxs:text-xxs xs:text-xs sm:text-sm lg:text-lg bg-[#A040A0]"
          >
            poison
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
